torch/testing/_internal/common_utils.py

```diff
@@ -1804,6 +1805,18 @@ def get_int64_dtype(dtype):
                                             dtype=int64_dtype, layout=layout, device=device, requires_grad=False),
                             int64_dtype, layout, device, fv + 5, False)
 
+# this helper method is to clone the input of test function
+# for the test file such as `test_ops.py` and `test_op_aliases.py`
+def clone_input_helper(input):
+    if isinstance(input, Sequence):
+        # use torch.clone for tensor, copy.deepcopy otherwise
+        return list(map(lambda x: torch.clone(x) if isinstance(x, torch.Tensor) else deepcopy(x),
+                        input))
+
+    if isinstance(input, torch.Tensor):
+        return torch.clone(input)
+    else:
+        return deepcopy(input)
```

Comment:
I think we can get away without copying non-tensor inputs, since I don't think we have any functions that edit non-tensor inputs? 

Ideally we'd also only clone when we're performing the operation inplace, but that can be a TODO for a later PR.

Reply:
Thanks for the advice, the `clone_input_helper` do not copy the non-tensor inputs in the latest PR.

Reply:
`test_ops.py` is updated to only clone tensor-type inputs for inplace operations.
```python
variant_forward = variant(*(clone_input_helper(input) if variant is inplace else input
                            for input in sample.input),
                          *sample.args,
                          **sample.kwargs)
self.assertEqual(variant_forward, expected_forward)
```